### PR TITLE
Adding pipeEnv.

### PR DESCRIPTION
**Pull request by rajkumar42**
Adding the option to provide environment variables to the pipe program.

### CODE REVIEW

**Comment by rajkumar42** [comment]
@chuckries @gregg-miskelly @caslan @wesrupert

File: src/MICore/LaunchOptions.cs [hunk 1]
```diff
@@ -109,6 +112,11 @@ static internal PipeLaunchOptions CreateFromXml(Xml.LaunchOptions.PipeLaunchOpti
         /// [Optional] Current working directory when the pipe program is invoked.
         /// </summary>
         public string PipeCwd { get; private set; }
+        
+        /// <summary>
+        /// [Optional] Enviroment variables for the pipe program.
+        /// </summary>
+        public EnvironmentEntry[] PipeEnvironment { get; private set; }
```
Comment (gregg-miskelly) [inline src/MICore/LaunchOptions.cs]: Can we make this a ReadOnlyCollection and/or IReadOnlyList. I don't like mutable public members of public types

File: src/MICore/Transports/PipeTransport.cs [hunk 2]
```diff
@@ -137,6 +137,11 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             proc.StartInfo.Arguments = pipeOptions.PipeArguments;
             proc.StartInfo.WorkingDirectory = pipeOptions.PipeCwd;
 
+            foreach (EnvironmentEntry entry in pipeOptions.PipeEnvironment)
+            {
+                proc.StartInfo.SetEnvironmentVariable(entry.Name, entry.Value);
```
Comment (gregg-miskelly) [inline src/MICore/Transports/PipeTransport.cs]: If I remember correctly (and maybe I don't) this is missing from Mono, and you want to use StartInfo.Environment.Add

**Comment by gregg-miskelly** [comment]
:shipit:

**Comment by wesrupert** [comment]
LGTM! :shipit:

File: src/MICore/Transports/PipeTransport.cs (hunk 2 above)
Comment (rajkumar42) [inline src/MICore/Transports/PipeTransport.cs]: I see StartInfo.SetEnvironmentVariable is used in LocalTransport.cs as well. 
@andrewcrawley @faxue-msft  can you please confirm?

Comment (andrewcrawley) [inline src/MICore/Transports/PipeTransport.cs]: Using StartInfo.SetEnvironmentVariable is correct - it's a helper we added that does the right thing for both CoreCLR and Mono.

LGTM.

**Pull request merged**